packages/vulcan-forms/lib/components/FormComponent.jsx

```diff
@@ -88,7 +100,15 @@ class FormComponent extends Component {
         value = merge({}, documentValue, currentValue);
       } else {
         // note: value has to default to '' to make component controlled
-        value = currentValue || documentValue || '';
+        //value = currentValue || documentValue || '';
+        // note: the previous line does not work when a checkbox is 'false' or a number is '0'
+        value = currentValue;
+        if (typeof value === 'undefined' || value === null) {
+          value = documentValue;
+        }
+        if (typeof value === 'undefined' || value === null) {
+          value = '';
+        }
```

Comment:
seems like something weird is going on here?

Reply:
So the original code was:
```
value = currentValue || documentValue || '';
```
But that does not work to set a checkbox to `false` or a number to `0`. So essentially the bug was that if you had a number input with the value "10" and then you typed "0" it would go back to 10. We can't use the || shorthand here.

Maybe I should use something other than `typeof value === 'undefined' || value === null`?

I spent many hours tracking these little bugs down.

Reply:
I meant you have the `if (typeof value === 'undefined' || value === null) {` line twice. Is that on purpose?

Reply:
Would this accomplish the same thing?

```
value = '';
        if (typeof currentValue !== 'undefined' && currentValue !== null) {
          value = currentValue;
        } else if (typeof documentValue !== 'undefined' && documentValue !== null) {
          value = documentValue;
        }
```

Reply:
Yes, I think so.